pkg/api/svc.go

```diff
@@ -117,6 +153,65 @@ func (s *Service) Handle(conn wasabi.Connection, r wasabi.Request) error {
 	}
 }
 
+// getRequestLimits is a helper function transform request limits mentioned in server configuration
+// into wasabi request limits, so that we can use the RateLimiter middleware.
+func getRequestLimits(rateLimitCfg RateLimits) (func(wasabi.Request) (string, time.Duration, uint64), error) {
+	if len(rateLimitCfg.Groups) == 0 {
+		return getDefaultRequestLimits(rateLimitCfg.General)
+	}
+
+	return getRateLimitForMethods(rateLimitCfg.Groups)
```

Comment:
Doesn't look right, we should just use `getRateLimitForMethods` only if you didn't find a group, use default rate limits if it's defined

Reply:
so what has priority when both are defined. lets say something llike this
```
general rate limit = 1000 request per 10s
per method rate limits
    - 100 chain request per 10s
    - 500 aggregate requests per 5s
```

in this case, what must be applied?

Reply:
@shan-96 if method is defined in a group, it should be rate limited by group, otherwise it should be rate limited by general limit

Reply:
@ksysoev but `requestLimitsFunc` is of type `(func(wasabi.Request) (string, time.Duration, uint64)`.

here method on request is not available only routing key

Reply:
does `request.Request` extend from `wasabi.Request`?

Reply:
request.Request is implementing wasabi.Request interface, https://github.com/ksysoev/deriv-api-bff/blob/main/pkg/core/request/request.go#L63 Routing key will be a method. as I said your `getRateLimitForMethods` is on the right track, we just need to handle scenario where we didn't manage to find methods in groups.

Reply:
wait a minute. If `routingKey == method` then how to find the property `group`

Reply:
@shan-96 when reading config create map method to group.